### PR TITLE
Added cluster support for cache.

### PR DESCRIPTION
**Pull request by rafalmnich**

### CODE REVIEW

File: cache/redis.go [hunk 1]
```diff
@@ -49,11 +49,34 @@ func NewRedis(uri string, opts ...RedisOptionsFunc) (Cache, error) {
...
+
+	c := redis.NewUniversalClient(uo)
+
+	return &redisCache{
+		client:  c,
+		decoder: stringDecoder{},
+	}, nil
+}
+
+func NewRedisUniversal(uri []string, opts ...RedisOptionsFunc) (Cache, error) {
```
Comment (nrwiersma) [inline cache/redis.go]: rename `uri` to `addrs` to be clear what you are passing in here.

File: cache/redis_internal_test.go [hunk 2]
```diff
@@ -45,7 +45,29 @@ func TestNewRedis(t *testing.T) {
 	c := v.(*redisCache)
 
 	assert.NoError(t, err)
-	assert.Equal(t, 12, c.client.Options().PoolSize)
+	clusterClient, isCluster := c.client.(*redis.ClusterClient)
+	if isCluster {
+		assert.Equal(t, 12, clusterClient.Options().PoolSize)
```
Comment (nrwiersma) [inline cache/redis_internal_test.go]: This is an impossible outcome to the test.

File: cache/redis_internal_test.go [hunk 3]
```diff
@@ -45,7 +45,29 @@ func TestNewRedis(t *testing.T) {
...
+	} else {
+		client, ok := c.client.(*redis.Client)
+		assert.True(t, ok)
+		assert.Equal(t, 12, client.Options().PoolSize)
+	}
+}
+
+func TestNewRedisUniversal(t *testing.T) {
+	v, err := NewRedisUniversal([]string{"redis://test", "redis://test2"}, WithPoolSize(12))
```
Comment (nrwiersma) [inline cache/redis_internal_test.go]: This is wrong. From the docs:
```
// Either a single address or a seed list of host:port addresses
// of cluster/sentinel nodes.
Addrs []string
```

File: cache/redis_internal_test.go [hunk 4]
```diff
@@ -45,7 +45,29 @@ func TestNewRedis(t *testing.T) {
...
+	c := v.(*redisCache)
+
+	assert.NoError(t, err)
+	clusterClient, isCluster := c.client.(*redis.ClusterClient)
+	if isCluster {
+		assert.Equal(t, 12, clusterClient.Options().PoolSize)
+	} else {
+		client, ok := c.client.(*redis.Client)
```
Comment (nrwiersma) [inline cache/redis_internal_test.go]: This is an impossible outcome for this test.

File: cache/redis_internal_test.go [hunk 5]
```diff
@@ -45,7 +45,29 @@ func TestNewRedis(t *testing.T) {
...
+		assert.Equal(t, 12, client.Options().PoolSize)
+	}
+}
+
+func TestNewRedisUniversal(t *testing.T) {
+	v, err := NewRedisUniversal([]string{"redis://test", "redis://test2"}, WithPoolSize(12))
+	c := v.(*redisCache)
+
+	assert.NoError(t, err)
+	clusterClient, isCluster := c.client.(*redis.ClusterClient)
```
Comment (nrwiersma) [inline cache/redis_internal_test.go]: Rather assert that it is indeed a cluster, as we expect it to be.

File: cache/redis_internal_test.go [hunk 6]
```diff
@@ -45,7 +45,29 @@ func TestNewRedis(t *testing.T) {
 	c := v.(*redisCache)
 
 	assert.NoError(t, err)
-	assert.Equal(t, 12, c.client.Options().PoolSize)
+	clusterClient, isCluster := c.client.(*redis.ClusterClient)
+	if isCluster {
+		assert.Equal(t, 12, clusterClient.Options().PoolSize)
+	} else {
+		client, ok := c.client.(*redis.Client)
```
Comment (nrwiersma) [inline cache/redis_internal_test.go]: Rather assert this is a non-clustered client, like we expect it to be.

File: cache/redis_internal_test.go [hunk 7]
```diff
@@ -45,7 +45,29 @@ func TestNewRedis(t *testing.T) {
...
+		assert.True(t, ok)
+		assert.Equal(t, 12, client.Options().PoolSize)
+	}
+}
+
+func TestNewRedisUniversal(t *testing.T) {
```
Comment (nrwiersma) [inline cache/redis_internal_test.go]: Please make a second test to compliment this one, for a non-clustered case.
This should become, `TestNewRedisUniversal_Cluster` and a new test `TestNewRedisUniversal_NonCluster` perhaps.

File: cache/redis_internal_test.go (hunk 2 above)
Comment (rafalmnich) [reply cache/redis_internal_test.go]: OK

File: cache/redis.go (hunk 1 above)
Comment (rafalmnich) [reply cache/redis.go]: OK

File: cache/redis_internal_test.go (hunk 3 above)
Comment (rafalmnich) [reply cache/redis_internal_test.go]: yeah, right.

File: cache/redis_internal_test.go (hunk 4 above)
Comment (rafalmnich) [reply cache/redis_internal_test.go]: true

File: cache/redis_internal_test.go (hunk 5 above)
Comment (rafalmnich) [reply cache/redis_internal_test.go]: ok

File: cache/redis_internal_test.go (hunk 6 above)
Comment (rafalmnich) [reply cache/redis_internal_test.go]: ok

File: cache/redis_internal_test.go (hunk 7 above)
Comment (rafalmnich) [reply cache/redis_internal_test.go]: OK

File: cache/redis_internal_test.go (hunk 2 above)
Comment (rafalmnich) [reply cache/redis_internal_test.go]: yeah

**Pull request merged**